lib/document/Document.js

```diff
@@ -165,20 +170,73 @@ class Document {
     return this;
   }
 
+  /**
+   * Get whole metadata object
+   *
+   * @return {Object}
+   */
+  getMetadata() {
+    return this.meta;
+  }
+
+  /**
+   * Set metadata object
+   *
+   * @param {Object} metaObject
+   * @return {Document}
+   */
+  setMetadata(metaObject) {
+    this.meta = {};
+
+    Object.entries(metaObject)
+      .forEach(([name, value]) => this.setMeta(name, value));
+
+    return this;
+  }
+
+  /**
+   * Retrieves the meta field specified by {path}
+   *
+   * @param {string} path
+   * @return {*}
+   */
+  getMeta(path) {
+    return lodashGet(this.meta, path);
+  }
+
+  /**
+   * Set the meta field specified by {path}
+   *
+   * @param {string} path
+   * @param {*} value
+   * @return {Document}
+   */
+  setMeta(path, value) {
+    lodashSet(this.meta, path, value);
+
+    return this;
+  }
+
   /**
    * Return Document as plain object
    *
    * @return {RawDocument}
    */
   toJSON() {
-    return {
+    const json = {
       $type: this.getType(),
       $scope: this.scope,
       $scopeId: this.scopeId,
       $rev: this.getRevision(),
       $action: this.getAction(),
       ...this.getData(),
     };
+
+    if (this.meta) {
```

Comment:
Is it work with an empty object? Probably better with `Object.keys(this.meta).length > 0`?